api/tests/unit/domain/services/certification/certification-service_test.js

```diff
@@ -0,0 +1,285 @@
+          createdAt: new Date('2017-12-23T15:23:12Z'),
+          firstName: 'Pumba',
+          lastName: 'De La Savane',
+          birthplace: 'Savane',
+          birthdate: '28/01/1992',
+          sessionId: 'MoufMufassa',
+          externalId: 'TimonsFriend',
+        }));
+        sinon.stub(assessmentResultRepository, 'get').resolves(null);
+      });
+
+      it('should return certification results with state at started, empty marks and undefined for information not yet valid', () => {
+        // given
+        const certificationCourseId = 1;
+
+        // when
+        const promise = certificationService.getCertificationResult(certificationCourseId);
+
+        // then
+        return promise.then((certification) => {
+          expect(certification.status).to.deep.equal('started');
+          expect(certification.competencesWithMark).to.deep.equal([]);
+          expect(certification.pixScore).to.deep.equal(undefined);
+          expect(certification.completedAt).to.deep.equal(undefined);
+          expect(certification.createdAt).to.deep.equal(new Date('2017-12-23T15:23:12Z'));
+          expect(certification.sessionId).to.deep.equal('MoufMufassa');
+        });
+      });
+
+      it('should know certification version', async () => {
+        // given
+        const certificationCourseId = 1;
+
+        // when
+        const certificationResult = await certificationService.getCertificationResult(certificationCourseId);
+
+        // then
+        expect(certificationResult.isV2Certification).to.be.false;
+      });
+    });
+  });
+
+  describe('#computeAnswersSuccessRate', () => {
+
+    context('when no answers is given', () => {
+
+      it('should have a trust level has unknown', () => {
+        // given
+        const answers = [];
+
+        // when
+        const successRate = certificationService._computeAnswersSuccessRate(answers);
+
+        // then
+        expect(successRate).to.equal(0);
+      });
+    });
+
+    context('when all answers are OK', () => {
+
+      it('should has a success rate at 100%', () => {
+        // given
+        const answers = [new Answer({ result: 'ok' }), new Answer({ result: 'ok' })];
+
+        // when
+        const successRate = certificationService._computeAnswersSuccessRate(answers);
+
+        // then
+        expect(successRate).to.equal(100);
+      });
+    });
+
+    context('when all answers are KO', () => {
+
+      it('should has a success rate at 0%', () => {
+        // given
+        const answers = [new Answer({ result: 'ko' }), new Answer({ result: 'ko' })];
+
+        // when
+        const successRate = certificationService._computeAnswersSuccessRate(answers);
+
+        // then
+        expect(successRate).to.equal(0);
+      });
+    });
+
+    context('when the answers are a mixed of valid and wrong answers', () => {
+
+      it('should has a success rate at 50% with 1W and 1R', () => {
```

Comment:
"it should have" On en retrouve quelques uns dans ce fichier

Reply:
Merci, je corrige